test/Toolchain.cpp

```diff
@@ -19,18 +19,93 @@
 #include <catch2/catch_test_macros.hpp>
 #include "SMCE/Toolchain.hpp"
 #include "defs.hpp"
+#include <iostream>
+#include <fstream>
 
 TEST_CASE("Toolchain invalid", "[Toolchain]") {
+    // Empty directory
     const auto path = SMCE_TEST_DIR "/empty_dir";
     std::filesystem::create_directory(path);
     smce::Toolchain tc{path};
-    REQUIRE(tc.check_suitable_environment());
     REQUIRE(tc.resource_dir() == path);
+    REQUIRE(tc.check_suitable_environment() == smce::toolchain_error::resdir_empty);
+    // Absent directory
+    const auto path2 = SMCE_TEST_DIR "/absent_dir";
+    smce::Toolchain tc2{path2};
+    REQUIRE(tc2.resource_dir() == path2);
+    REQUIRE(tc2.check_suitable_environment() == smce::toolchain_error::resdir_absent);
+    // Directory is file
+    const auto dir3 = SMCE_TEST_DIR "/dir_is_file";
+    std::filesystem::create_directory(dir3);
+    const auto dirfile = SMCE_TEST_DIR "/dir_is_file/file.txt";
+    std::ofstream output(dirfile);
+    smce::Toolchain tc3{dirfile};
+    REQUIRE(tc3.resource_dir() == dirfile);
+    REQUIRE(tc3.check_suitable_environment() == smce::toolchain_error::resdir_file);
+    // Invalid sketch path
+    smce::Toolchain tc4{SMCE_PATH};
+    REQUIRE(!tc4.check_suitable_environment());
+    smce::Sketch sk{SKETCHES_PATH "invalid", {.fqbn = "arduino:avr:nano"}};
+    const auto ec4 = tc4.compile(sk);
+    REQUIRE(ec4 == smce::toolchain_error::sketch_invalid);
+    // Invalid plugin
+    smce::Toolchain tc5{SMCE_PATH};
+    REQUIRE(!tc5.check_suitable_environment());
+    // clang-format off
+    smce::PluginManifest invalid_pl{
+        ".",
+        "0.0",
+        {},
+        {},
+        "..",
+        "file://" PATCHES_PATH "invalid",
+        smce::PluginManifest::Defaults::arduino,
+        {},
+        {},
+        {},
+        {}
+    };
+
+    smce::SketchConfig skc{
+        "arduino:avr:nano",
+        {},
+        { smce::SketchConfig::ArduinoLibrary{"."} },
+        { std::move(invalid_pl) }
+    };
+    // clang-format on
+    smce::Sketch sk5{SKETCHES_PATH "patch", std::move(skc)};
+    const auto ec5 = tc5.compile(sk5);
+    REQUIRE(ec5 == smce::toolchain_error::invalid_plugin_name);
```

Comment:
Instead of using `var8` and other kinds of numbered variables, it would be much cleaner to use local scopes.
In fact, what you're saying with your in-code comments suggests that those comments should be gone and replaced by appropriately named [`SECTION`s](https://github.com/catchorg/Catch2/blob/devel/docs/tutorial.md#test-cases-and-sections) (which have their own scope thus solves the variable naming problem)

Reply:
ahhh yes, thanks that's a great solution! will divide the tests up in sections